src/Features/CSharp/Portable/Completion/CompletionProviders/TypeImportCompletionProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal class TypeImportCompletionProvider : AbstractTypeImportCompletionProvider
+    {
+        internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
+            => CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
```

Comment:
when one should use CompletionUtilities.IsTriggerCharacter? is there a particular reason it is chosen for this provider? I am looking through IsInsertionTrigger and not sure which one should be used when.

Reply:
@heejaechang this should basically follow the pattern of SymbolCompletion.

Reply:
and, it's possibl we might want to share code between these guys since they should be very similar.